packages/dashboard/src/managers/trajectory-socket-manager.ts

```diff
@@ -0,0 +1,51 @@
+import EventEmitter from 'eventemitter3';
+
+type Events = {
+  updated: [];
+};
+
+export default class TrajectorySocketManager extends EventEmitter<Events> {
+  private _ongoingRequest: Promise<MessageEvent> | null = null;
+
+  private _listenOnce<K extends keyof WebSocketEventMap>(
+    event: K,
+    _webSocket: WebSocket,
+    listener: (e: WebSocketEventMap[K]) => unknown,
+  ): void {
+    _webSocket.addEventListener(event, (e) => {
+      _webSocket.removeEventListener(event, listener);
+      listener(e);
+    });
+  }
+
+  /**
+   * Sends a message and waits for response from the server.
+   *
+   * @remarks This is an alternative to the old implementation of creating a promise, storing the
+   * resolver and processing each message in an event loop. Advantage of this is that each message
+   * processing logic can be self-contained without a need for a switch or if elses.
+   */
+  async _send(
```

Comment:
public methods shouldn't be prefixed with _

Reply:
If this method is only intended for use in the child class, do you think marking it with an `_` as a reminder would be a good idea? I think I read an article before and the author does this, but I think it is his preference

Reply:
You can make it `protected` in this case. Also `_websocket` shouldn't have a _ prefix as well.